File: app/src/main/java/com/example/fbproj1/ImageAdapterGridView.java
package com.example.fbproj1;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.GridView;
import android.widget.ImageView;

import java.util.ArrayList;

public class ImageAdapterGridView extends BaseAdapter {
    private Context context;
    private ArrayList<PhotoCardItem> photoCollection;

    public ImageAdapterGridView(Context context, ArrayList<PhotoCardItem> photoCollection) {
        this.context = context;
        this.photoCollection = photoCollection;
    }

    public int getCount() {
        return photoCollection.size();
    }

    public Object getItem(int position) {
        return position;
    }

    public long getItemId(int position) {
        return position;
    }

    public View getView(int position, View convertView, ViewGroup parent) {
        ImageView imageView;
        if (convertView == null) {
            imageView = new ImageView(context);
            imageView.setLayoutParams(new GridView.LayoutParams(300, 300));
            imageView.setScaleType(ImageView.ScaleType.CENTER_CROP);
            imageView.setPadding(5, 5, 5, 5);
        } else {
            imageView = (ImageView) convertView;
        }
        imageView.setImageBitmap(photoCollection.get(position).getPciBitmapDataObj().getCurrentImage());
        return imageView;
    }

}

File: app/src/main/java/com/example/fbproj1/ImageAdapterGridView.kt
package com.example.fbproj1

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.GridView
import android.widget.ImageView

class ImageAdapter(private val context: Context) : BaseAdapter() {

    //---returns the number of images---
    override fun getCount(): Int {
        return imageIDs.length
    }

    //---returns the ID of an item---
    override fun getItem(position: Int): Any {
        return position
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    //---returns an ImageView view---
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val imageView: ImageView
        if (convertView == null) {
            imageView = ImageView(context)
            imageView.layoutParams = GridView.LayoutParams(185, 185)
            imageView.scaleType = ImageView.ScaleType.CENTER_CROP
            imageView.setPadding(5, 5, 5, 5)
        } else {
            imageView = convertView as ImageView?
        }
        imageView.setImageResource(imageIDs[position])
        return imageView
    }
}